articles/clients/client-grant-types.md

```diff
@@ -0,0 +1,135 @@
+* Specification-conformant (such as OpenID Connect);
+
+The following is a list of Auth0 extension/OIDC specification-conformant `grant_types` that currently exist for use in Auth0:
+
+* `implicit`
+* `authorization_code`
+* `client_credentials`
+* `password`
+* `refresh_token`
+* `http://auth0.com/oauth/grant-type/password-realm`
+* `http://auth0.com/oauth/grant-type/mfa-oob`
+* `http://auth0.com/oauth/grant-type/mfa-otp`
+* `http://auth0.com/oauth/grant-type/mfa-recovery-code`
+
+The following is a list of legacy `grant_types`:
+
+* `http://auth0.com/oauth/legacy/grant-type/ro`
+* `http://auth0.com/oauth/legacy/grant-type/ro/jwt-bearer`
+* `http://auth0.com/oauth/legacy/grant-type/delegation/refresh_token`
+* `http://auth0.com/oauth/legacy/grant-type/delegation/id_token`
+* `http://auth0.com/oauth/legacy/grant-type/access_token`
+
+## Edit the `grant_type` Property
+
+To edit the `grant_type` property for your Auth0 Client, you'll need to make the appropriate call to the [Update a Client endpoint](/api/management/v2#!/Clients/patch_clients_by_id) of the [Management API](/api/management/v2).
+
+:::panel-warning Legacy Grant Types
+As of TBD, new Auth0 customers **cannot** add *any* of the legacy grant types to their Clients. Only customers as of TBD can add legacy grant types to their existing Clients.
+:::
+
+Auth0 requires the `grant_type` property. Attempting to use *any* flow with a Client lacking the appropriate `grant_type` for that flow (or with the field empty) will result in the following error:
+
+```text
+Grant type `grant_type` not allowed for the client.
+```
+
+### Existing Clients
+
+To avoid changes in functionality for current Auth0 customers, we will populate the `grant_types` property for all existing Clients as of TBD with **all** Auth0 legacy, Auth0 extension, and specification-conformant grant types.
+
+### New Clients
+
+Depending on whether a newly-created Client is **public** or **confidential**, the Client will have varying access to grant types.
+
+### Public Clients
+
+Public Clients, indicated by the `token_endpoint_auth_method` flag set to `none`, are those created in the Dashboard for Native and Single Page Applications. They can be set to the following grant types:
+
+* `implicit`;
+* `authorization_code`;
+* `refresh_token`.
+
+### Confidential Clients
+
+Confidential Clients, indicated by the `token_endpoint_auth_method` flag set to anything *except* `none`, are those created in the Dashboard for Regular Web Applications or Non-Interactive Clients. Additionally, any Client where `token_endpoint_auth_method` is unspecified is confidential. Confidential Clients can be set to the following grant types:
+
+
+* `implicit`;
+* `authorization_code`;
+* `refresh_token`;
+* `client_credentials`.
+
+### Trusted First-Party Clients
+
+When creating a new Client via the [Management Dashboard](${manage_url}), Auth0 sets the `is_first_party` flag to `true`.
+
+The Client can therefore use any of the following grant types:
```

Comment:
Trusted first party clients can additionally use the following `grant_types`: 

* `password`
* `http://auth0.com/oauth/grant-type/password-realm`
* `http://auth0.com/oauth/grant-type/mfa-oob`
* `http://auth0.com/oauth/grant-type/mfa-otp`
* `http://auth0.com/oauth/grant-type/mfa-recovery-code`